### PR TITLE
Add extract function to facilitate agnostic cube input orders to CLIs

### PR DESCRIPTION
**Pull request by Kat-90**
This PR adds functionality to simple order specific CLI's to be able to pass input cubes agnostically.
It also adds input cube functions to the corresponding plugins where necessary.

This PR acts as a first step toward a bigger issue: #1770 and more PR's will be put up as plugins/CLI's are changed.

Addresses #GitHubissuenum

Description

Testing:
 - [X] Ran tests and they passed OK
 - [ ] Added new tests for the new feature(s)

CLA
 - [ ] If a new developer, signed up to CLA

### CODE REVIEW

**Review by gavinevans — COMMENTED** [review]
Thanks @Kat-90 👍 

I think that these changes look correct. I think in this instance, you don't need to make use of `get_model_id_attr`, as you've done, as `generate_mandatory_attributes` is already doing the desired attribute check. 

There is one [mailmap](https://github.com/metoppv/improver/runs/7634244432?check_suite_focus=true#step:3:93) error that has occurred. Let's have a quick chat about this.

File: improver/cli/wind_gust_diagnostic.py [hunk 1]
```diff
@@ -64,9 +64,14 @@ def process(
...
+    from iris.cube import CubeList
+
     from improver.wind_calculations.wind_gust_diagnostic import WindGustDiagnostic
 
-    result = WindGustDiagnostic(wind_gust_percentile, wind_speed_percentile)(
+    wind_gust, wind_speed = CubeList(cubes).extract(
+        ["wind_speed_of_gust", "wind_speed"]
+    )
+
+    return WindGustDiagnostic(wind_gust_percentile, wind_speed_percentile)(
```
Comment (gavinevans) [inline improver/cli/wind_gust_diagnostic.py:75]: I think that the WindGustDiagnostic plugin could also be updated with the usage of the    `assert_spatial_coords_match` and `assert_time_coords_valid` functions. I can see that this plugin is already comparing whether the time coordinates are valid on [these lines](https://github.com/metoppv/improver/blob/master/improver/wind_calculations/wind_gust_diagnostic.py#L185-L202) by checking whether the wind speed time point is within the time bounds provided by the wind gust cube. This kind of check isn't supported by `assert_time_coords_valid` but we could consider whether `assert_time_coords_valid` could be updated to support this kind of check.

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks @Kat-90 👍 

I've added a couple of comments.

**Comment by Kat-90** [comment]
NB: I have left out the assert_time_coords function from wind_gust_diagnostic.py as this plugin takes one cube with time points (wind_speed) and one cube with time bounds (wind_gust).

**Pull request closed without merging**